server/athenian/api/internal/features/entries.py

```diff
@@ -206,13 +209,113 @@ def _align_time_min_max(
             return self.align_time_min_max(time_intervals, 100500)
         return self.align_time_min_max(time_intervals, self._quantile_stride)
 
+    @sentry_span
+    @cached(
+        exptime=short_term_exptime,
+        serialize=pickle.dumps,
+        deserialize=pickle.loads,
+        key=lambda requests, quantiles, labels, jira, exclude_inactive, bots, release_settings, logical_settings, **kwargs: (  # noqa
+            ",".join(str(req) for req in requests),
+            ",".join(str(q) for q in quantiles),
+            labels,
+            jira,
+            exclude_inactive,
+            release_settings,
+            logical_settings,
+        ),
+        cache=lambda self, **_: self._cache,
+    )
+    async def batch_calc_pull_request_metrics_line_github(
+        self,
+        requests: Sequence[PullRequestMetricsLineRequest],
+        quantiles: Sequence[float | int],
+        labels: LabelFilter,
+        jira: JIRAFilter,
+        exclude_inactive: bool,
+        bots: set[str],
+        release_settings: ReleaseSettings,
+        logical_settings: LogicalRepositorySettings,
+        prefixer: Prefixer,
+        branches: pd.DataFrame,
+        default_branches: dict[str, str],
+        fresh: bool,
+    ) -> Sequence[np.ndarray]:
+        """Execute a set of requests for pull request metrics.
+
+        A numpy array is returned for every request in `requests`.  The numpy array has the same
+        format of the return type of `calc_pull_request_metrics_line_github()`.
+
+        Calling this method with multiple `requests` is more efficient than calling
+        `calc_pull_request_metrics_line_github()` multiple times.
+
+        """
+        all_repositories = set(
+            chain.from_iterable(chain.from_iterable(request.repositories) for request in requests),
+        )
+        all_participants = _merge_participants(
+            list(chain.from_iterable(request.participants for request in requests)),
+        )
+        all_intervals = list(chain.from_iterable(request.time_intervals for request in requests))
+        all_metrics = set(chain.from_iterable(request.metrics for request in requests))
+
+        time_from, time_to = self._align_time_min_max(all_intervals, quantiles)
+        df_facts = await self.calc_pull_request_facts_github(
+            time_from,
+            time_to,
+            all_repositories,
+            all_participants,
+            labels,
+            jira,
+            exclude_inactive,
+            bots,
+            release_settings,
+            logical_settings,
+            prefixer,
+            fresh,
+            need_jira_mapping(all_metrics),
```

Comment:
it is worth renaming `from x import need_jira_mapping as pr_metrics_need_jira_mapping`